lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_model_base.rb

```diff
@@ -60,7 +60,7 @@ def self.ar_model_associations
 
     def self.expose_class_attributes(subclass)
       subclass.class_eval do
-        model.attribute_names.each do |attr|
+        (model.attribute_names | model.try(:attribute_aliases)&.keys).each do |attr|
```

Comment:
Nice find on this location.  Does it make sense to have method that virtual attributes adds that provides this?

I see we have similar code here and https://github.com/ManageIQ/manageiq-api/pull/1278.  Perhaps, we could have something in core that extends AR to give us `exposed_attribute_names` so we can have one place to test that `exposed_attribute_names` includes virtual attributes, columns, aliased attributes, etc.  Perhaps deprecated_attributes are in the same bucket.

Reply:
@jrafanie Did you see my comment above?

Reply:
see above